### PR TITLE
RavenDB-13584 Adding a test for node distribution when node goes down

### PR DESCRIPTION
**Pull request by talweiss1982**
Also added operation for setting the dynamic redistribution feature

### CODE REVIEW

File: test/SlowTests/Cluster/NodeRedistribution.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            var notInDbGroup = cluster.Nodes.First(s => res.Servers.Contains(s) == false);
+            using (var store = new DocumentStore
+            {
+                Database = databaseName,
+                Urls = new[] { cluster.Leader.WebUrl }
+            }.Initialize())
+            {
+                store.Maintenance.Server.Send(new SetDatabaseDynamicDistribution(store.Database, true));
+                await DisposeAndRemoveServer(res.Servers.First());     
+                Thread.Sleep((moveToRehabGraceTime + replicaTimeout)* 2 * 1000);
```
Comment (ayende) [inline test/SlowTests/Cluster/NodeRedistribution.cs]: Can we avoid the sleep somehow? Can we do an active check?
This is likely to fail if the cluster observer is a bit slow.

**Review by ayende — CHANGES_REQUESTED** [review]

File: test/SlowTests/Cluster/NodeRedistribution.cs (hunk 1 above)
Comment (talweiss1982) [reply test/SlowTests/Cluster/NodeRedistribution.cs]: I can change it if you want

Comment (ayende) [reply test/SlowTests/Cluster/NodeRedistribution.cs]: please do

Comment (talweiss1982) [reply test/SlowTests/Cluster/NodeRedistribution.cs]: now it seems to fail on debug

Comment (ayende) [reply test/SlowTests/Cluster/NodeRedistribution.cs]: why?

Comment (talweiss1982) [reply test/SlowTests/Cluster/NodeRedistribution.cs]: not sure why

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using System.Net.Http;
+using System.Text;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.ServerWide.Operations
+{
+    public class SetDatabaseDynamicDistribution : IServerOperation<bool>
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: rename to `SetDatabaseDynamicDistributionOperation`

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        private bool _allowDynamicDistribution;
+        private string _databaseName;
+
+        public SetDatabaseDynamicDistribution(string databaseName, bool allowDynamicDistribution)
+        {
+            _allowDynamicDistribution = allowDynamicDistribution;
+            _databaseName = databaseName;
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: Let's assert if this is not null or empty

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        public RavenCommand<bool> GetCommand(DocumentConventions conventions, JsonOperationContext context)
+        {
+            return new SetDatabaseDynamicDistributionCommand(_databaseName, _allowDynamicDistribution);
+        }
+    }
+
+    public class SetDatabaseDynamicDistributionCommand : RavenCommand<bool>, IRaftCommand
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: mark as private and put inside the operation class

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs (hunk 2 above)
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: Do we want to expose this as a part of public API? Maybe internal is enough?

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs (hunk 3 above)
Comment (talweiss1982) [reply src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: what is null?

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs (hunk 2 above)
Comment (talweiss1982) [reply src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: why not? I think it was just missing and it is not breaking anything

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: databaseName, string

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistributionOperation.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+using System.Net.Http;
+using System.Text;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.ServerWide.Operations
+{
+    public class SetDatabaseDynamicDistributionOperation : IServerOperation<bool>
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistributionOperation.cs]: `IServerOperation<bool>` needs to be changed to `IServerOperation`
The endpoint does not return anything

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistribution.cs]: ok

File: src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistributionOperation.cs (hunk 5 above)
Comment (talweiss1982) [reply src/Raven.Client/ServerWide/Operations/SetDatabaseDynamicDistributionOperation.cs]: I missed that interface, was looking for it, was below the other one

**Review by ppekrol — APPROVED** [review]

**Pull request merged**